src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -603,6 +619,30 @@ internal override CSharpAttributeData EarlyDecodeWellKnownAttribute(ref EarlyDec
                 {
                     arguments.GetOrCreateData<ParameterEarlyWellKnownAttributeData>().HasCallerMemberNameAttribute = true;
                 }
+                else if (CSharpAttributeData.IsTargetEarlyAttribute(arguments.AttributeType, arguments.AttributeSyntax, AttributeDescription.CallerArgumentExpressionAttribute))
```

Comment:
As the language exists today we can do this as an early attribute, but if we do implement the nameof proposal (allowing `nameof(parameter)` in parameters/attributes on methods) I don't believe it will end well (this will cause a loop or fail to bind entirely). Please add a prototype comment to consider moving binding of the actual attribute data into regular attribute binding.

Reply:
@333fred To make sure I understand this correctly, this is about `nameof` binding being done *after* we are trying to get the parameter name?

This also affects existing code, for example:

https://github.com/dotnet/roslyn/blob/840875004775efb54881af9e5dbc2d79e3f71f44/src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs#L898-L912

Reply:
>  To make sure I understand this correctly, this is about nameof binding being done after we are trying to get the parameter name?

I'm not sure I understand the question. My concern here is:

1. We're attempting to bind the parameter
2. The parameter has this attribute, and uses nameof on a parameter in this method (even itself!)
3. Attempting to bind the nameof argument causes us to bind the method, which causes us to bind the nameof, which causes us to bind the method, which... etc.

Today, the parameter is not in scope for that `nameof`, so our standard early attribute handling will work correctly without a chance for a loop. I'm not certain it will work correctly in the face of an updated `nameof`, however, so to do the right thing we'd want to move binding of the attribute argument to the standard binding location, which isn't susceptible to this problem.

Reply:
I think I got the general idea here. I think (most likely) this will require fixes in other existing areas first (e.g, the ConditionalAttribute above), so I'll probably follow the same work and add tests.